File: iosApp/iosApp/Presentation/Theme/AppBrandTheme.swift
//
//  AppBrandTheme.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/07/08.
//

import Foundation
import Shared
import SwiftUI

enum AppBrandTheme: String, CaseIterable, Identifiable {
    case maidCafe
    case mensConCafe

    var id: String { rawValue }

    var title: String {
        switch self {
        case .maidCafe:
            return String(localized: String.LocalizationValue("settings_brand_theme_maid"), table: "Localizable")
        case .mensConCafe:
            return String(localized: String.LocalizationValue("settings_brand_theme_mens"), table: "Localizable")
        }
    }

    init(brandTheme: BrandTheme) {
        switch brandTheme {
        case .mensConCafe:
            self = .mensConCafe
        default:
            self = .maidCafe
        }
    }

    var sharedBrandTheme: BrandTheme {
        switch self {
        case .maidCafe:
            return .maidCafe
        case .mensConCafe:
            return .mensConCafe
        }
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/theme/AppBrandTheme.kt
package com.hhp227.concafe.presentation.theme

import com.hhp227.concafe.domain.model.BrandTheme

enum class AppBrandTheme {
    MAID_CAFE,
    MENS_CON_CAFE
}

fun BrandTheme.toPresentationBrandTheme(): AppBrandTheme {
    return when (this) {
        BrandTheme.MAID_CAFE -> AppBrandTheme.MAID_CAFE
        BrandTheme.MENS_CON_CAFE -> AppBrandTheme.MENS_CON_CAFE
    }
}

fun AppBrandTheme.toDomainBrandTheme(): BrandTheme {
    return when (this) {
        AppBrandTheme.MAID_CAFE -> BrandTheme.MAID_CAFE
        AppBrandTheme.MENS_CON_CAFE -> BrandTheme.MENS_CON_CAFE
    }
}
